wandb/sdk/lib/sock_client.py

```diff
@@ -45,13 +50,32 @@ def shutdown(self, val: int) -> None:
     def set_socket(self, sock: socket.socket) -> None:
         self._sock = sock
 
+    def _sendall_with_timeout(self, data: bytes) -> None:
```

Comment:
this name is a bit misrepresented... there is a potential timeout

maybe sendall_with_retry and a comment would be better

Reply:
but `sendall` already has a retry... i was trying to imply that this handles a timeout exception

Reply:
yep... but to be "with_timeout" suggests to me that this code will send with timeout... and in the normal case it does not.

the overly verbose name would be "sendall_and_handle_possible_timeout".   
but there is probably something shorter :)